src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -80,6 +94,137 @@ def spark_version_compatibility(spark_version: str) -> str:
     return "supported"
 
 
+class AzureServicePrincipalCrawler(CrawlerBase):
+    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
+        super().__init__(sbe, "hive_metastore", schema, "azure_service_principals")
+        self._ws = ws
+        self._azure_spn_list_with_data_access = []
```

Comment:
```suggestion
```
do not rely on a private class property, but instead - rely on return values of the methods.